### ISSUE TITLE
timedatectl: caches NTP enablement state, better shouldn't

### ISSUE THREAD

**Issue by adrelanos** [systemd/systemd#1329]
Created a drop-in file to disable systemd-timesyncd.

`cat /lib/systemd/system/systemd-timesyncd.service.d/40_off.conf`

```
[Unit]
ConditionPathExists=!/lib/systemd/system/systemd-timesyncd.service.d/40_off.conf
```

`timedatectl set-ntp true`

It works as expected.

`sudo systemctl status systemd-timesyncd.service`

```
● systemd-timesyncd.service - Network Time Synchronization
   Loaded: loaded (/lib/systemd/system/systemd-timesyncd.service; enabled)
  Drop-In: /lib/systemd/system/systemd-timesyncd.service.d
           └─40_off.conf
   Active: inactive (dead)
           start condition failed at Tue 2015-09-22 10:54:47 UTC; 29s ago
           ConditionPathExists=!/lib/systemd/system/systemd-timesyncd.service.d/40_off.conf was not met
     Docs: man:systemd-timesyncd.service(8)
```

But timedatectl still reports ntp enabled even if it is not.

`timedatectl`

```
      Local time: Tue 2015-09-22 10:55:27 UTC
  Universal time: Tue 2015-09-22 10:55:27 UTC
        RTC time: n/a
       Time zone: Etc/UTC (UTC, +0000)
     NTP enabled: yes
NTP synchronized: yes
 RTC in local TZ: no
      DST active: n/a
```

**Comment by poettering** [comment]
Creating a condition that fails is not disabling a unit. The unit is still enabled, as is shown in the "systemctl status" output. To disable a unit use "systemctl disable"...

What's the rationale for that weird condition drop-in?

**Comment by adrelanos** [comment]
> What's the rationale for that weird condition drop-in?

Just for demonstration purposes.
Conflicting with another time synchronization daemon.

**Comment by adrelanos** [comment]
systemd-timesyncd gets enabled by default on Debian jessie?
(File `/etc/systemd/system/sysinit.target.wants/systemd-timesyncd.service` exists.)

**Comment by poettering** [comment]
We recommend distros enable it by default but other NTP implementation should have Conflicts= on it, so that the other implementations win if both are enabled.

**Comment by adrelanos** [comment]
Even after `sudo systemctl disable systemd-timesyncd`, `timedatectl` reports ntp still being enabled.

```
~ $ timedatectl
      Local time: Tue 2015-09-22 11:34:30 UTC
  Universal time: Tue 2015-09-22 11:34:30 UTC
        RTC time: n/a
       Time zone: Etc/UTC (UTC, +0000)
     NTP enabled: no
NTP synchronized: yes
 RTC in local TZ: no
      DST active: n/a
~ $ timedatectl set-ntp 1
~ $ timedatectl
      Local time: Tue 2015-09-22 11:34:31 UTC
  Universal time: Tue 2015-09-22 11:34:31 UTC
        RTC time: n/a
       Time zone: Etc/UTC (UTC, +0000)
     NTP enabled: yes
NTP synchronized: yes
 RTC in local TZ: no
      DST active: n/a
~ $ sudo systemctl disable systemd-timesyncd
Removed symlink /etc/systemd/system/sysinit.target.wants/systemd-timesyncd.service.
~ $ sudo systemctl status systemd-timesyncd
● systemd-timesyncd.service - Network Time Synchronization
   Loaded: loaded (/lib/systemd/system/systemd-timesyncd.service; disabled)
  Drop-In: /lib/systemd/system/systemd-timesyncd.service.d
           └─40_off.conf
   Active: inactive (dead)
     Docs: man:systemd-timesyncd.service(8)

Sep 22 10:51:49 work systemd[1]: Started Network Time Synchronization.
Sep 22 10:51:49 work systemd-timesyncd[1873]: Using NTP server 85.114.130.211:123 (0.debian.pool.ntp.org).
Sep 22 10:51:49 work systemd-timesyncd[1873]: interval/delta/delay/jitter/drift 64s/-0.002s/0.034s/0.000s/-13ppm
Sep 22 10:52:53 work systemd-timesyncd[1873]: interval/delta/delay/jitter/drift 128s/+0.001s/0.034s/0.000s/-11ppm
Sep 22 10:54:38 work systemd[1]: Stopping Network Time Synchronization...
Sep 22 10:54:38 work systemd[1]: Stopped Network Time Synchronization.
Sep 22 10:54:47 work systemd[1]: Started Network Time Synchronization.
Sep 22 11:30:11 work systemd[1]: Started Network Time Synchronization.
Sep 22 11:33:22 work systemd[1]: Stopped Network Time Synchronization.
Sep 22 11:34:31 work systemd[1]: Started Network Time Synchronization.
~ $ timedatectl
      Local time: Tue 2015-09-22 11:34:31 UTC
  Universal time: Tue 2015-09-22 11:34:31 UTC
        RTC time: n/a
       Time zone: Etc/UTC (UTC, +0000)
     NTP enabled: yes
NTP synchronized: yes
 RTC in local TZ: no
      DST active: n/a
```

**Comment by poettering** [comment]
Either way, what you are doing is not "disabled", it's just "enabled, but a condition isn't met", hence the timediatectl output is correct.

**Comment by poettering** [comment]
Ah, our new comments crossed each other on the line... Reopening.

**Comment by adrelanos** [comment]
Looks like I have a different expectation of what 'NTP enabled' means.

**Comment by poettering** [comment]
It appears that timedated is caching stuff it shouldn't cache...

**Comment by poettering** [comment]
@adreianos what timedatectl thinks "NTP enabled" means is documented under "set-ntp" in the man page...

**Comment by poettering** [comment]
So the there's a bug in timedated, that it caches the state where it shouldn't. Also the man page is a bit confusing, as it should really say that it's not just about enabling the unit, but also starting it...

**Comment by adrelanos** [comment]
Following the principle of the least surprise, I suggest to change the string in `timedatectl` from `NTP enabled` to `systemd-timesyncd enabled`.

**Comment by adrelanos** [comment]
...Because NTP is a well established project name. By changing that string, no expectation of disabling other NTP would be created. And therefore nothing to be noted in the man page.

**Comment by poettering** [comment]
It doesn't say "NTP" since a while. It says "Network time on" since 2fc4f5bd924b9732afc70f8a6da80573f833fc9d

**Comment by poettering** [comment]
which version of systemd are you using anyway?

**Comment by adrelanos** [comment]
```
systemd --version
systemd 215
+PAM +AUDIT +SELINUX +IMA +SYSVINIT +LIBCRYPTSETUP +GCRYPT +ACL +XZ -SECCOMP -APPARMOR
```

**Comment by poettering** [comment]
Please do not report bugs against such old systemd versions upstream. Only report bugs in the two most recent releases upstream, and everything else in the distro. Current is 226, that's more than 10 releases newer!

**Comment by adrelanos** [comment]
It's the version that comes with Debian stable. https://packages.debian.org/jessie/systemd

**Comment by johannbg** [comment]
You will need to test with the latest upstream version of systemd ( which is basically what every upstream will tell you when you file bug upstream, test with the latest version of the component ) and or file bugs downstream and have the downstream maintainers deal with it since this bug might already be fixed upstream and or is only relevant to the downstream implementation of systemd ( or any other component for that matter )

**Comment by adrelanos** [comment]
Alright.

**Comment by yuwata** [comment]
Fix is waiting in PR #8615.
